connector/oidc/oidc.go

```diff
@@ -87,6 +87,27 @@ type Config struct {
 		// Configurable key which contains the groups claims
 		GroupsKey string `json:"groups"` // defaults to "groups"
 	} `json:"claimMapping"`
+
+	// ClaimModifications holds all claim modifications options
+	ClaimModifications struct {
+		NewGroupsFromClaims []NewGroupsFromClaims `json:"newGroupsFromClaims"`
+	} `json:"claimModifications"`
+}
+
+// List of groups claim elements to create by concatenating other claims
+type NewGroupsFromClaims struct {
+	// List of claim to join together
+	ClaimList []string `json:"claimList"`
```

Comment:
```suggestion
	Claims []string `json:"claims"`
```

Maybe just call this "claims"?